src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -424,3753 +416,3744 @@ private void BuildApplicationInitializerBody(IndentedStringBuilder writer, XamlO
+			=> IsMemberInside(xamlObject, "DataTemplate")
+			|| IsMemberInside(xamlObject, "ControlTemplate")
+			|| IsMemberInside(xamlObject, "ItemsPanelTemplate");
+
+		private bool IsMemberInsideResourceDictionary(XamlObjectDefinition xamlObject)
+			=> IsMemberInside(xamlObject, "ResourceDictionary", maxDepth: 1);
+
+		private static bool IsMemberInside(XamlObjectDefinition xamlObject, string typeName, int? maxDepth = null)
+		{
+			if (xamlObject == null)
+			{
+				return false;
+			}
+
+			int depth = 0;
+			do
+			{
+				if (xamlObject.Type?.Name == typeName)
+				{
+					return true;
+				}
+
+				xamlObject = xamlObject.Owner;
+			}
+			while (xamlObject != null && (maxDepth == null || ++depth <= maxDepth));
+
+			return false;
+		}
+
+		/// <summary>
+		/// Inserts explicit cast if a resource is being assigned to a property of a different type
+		/// </summary>
+		private string GetCastString(INamedTypeSymbol targetProperty, XamlObjectDefinition targettingValue)
+		{
+			if (targetProperty != null
+				&& targetProperty.Name != targettingValue?.Type.Name
+				)
+			{
+				return $"({targetProperty.ToDisplayString()})";
+			}
+			return "";
+		}
+
+		/// <summary>
+		/// Gets the key referred to by a StaticResourceExtension or ThemeResourceExtension.
+		/// </summary>
+		/// <param name="member">The StaticResourceExtension or ThemeResourceExtension member</param>
+		/// <returns>(Key referred to, true if ThemeResource)</returns>
+		private (string key, bool isThemeResourceExtension) GetStaticResourceKey(XamlMemberDefinition member)
+		{
+			var staticResourceNode = member.Objects.FirstOrDefault(o => o.Type.Name == "StaticResource");
+			var themeResourceNode = member.Objects.FirstOrDefault(o => o.Type.Name == "ThemeResource");
+
+			var staticResourcePath = staticResourceNode?.Members.First().Value.ToString();
+			var themeResourcePath = themeResourceNode?.Members.First().Value.ToString();
+
+			return (staticResourcePath ?? themeResourcePath, themeResourceNode != null);
+		}
+
+		/// <summary>
+		/// Returns code for simple initialization-time retrieval for StaticResource/ThemeResource markup.
+		/// </summary>
+		private string GetSimpleStaticResourceRetrieval(XamlMemberDefinition member, INamedTypeSymbol targetPropertyType = null)
+		{
+			//Both ThemeResource and StaticResource use the same lookup mechanism
+			var resourcePath = GetStaticResourceKey(member).key;
+
+			if (resourcePath == null)
+			{
+				return null;
+			}
+			else
+			{
+				targetPropertyType = targetPropertyType ?? FindPropertyType(member.Member);
+
+				return GetSimpleStaticResourceRetrieval(targetPropertyType, resourcePath);
+			}
+		}
+
+		/// <summary>
+		/// Returns code for simple initialization-time retrieval for StaticResource/ThemeResource markup.
+		/// </summary>
+		private string GetSimpleStaticResourceRetrieval(INamedTypeSymbol targetPropertyType, string keyStr)
+		{
+			var directProperty = GetResourc
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this empty statement.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305744)